### PR TITLE
New Easter eggs which might actually be useful...

### PR DESCRIPTION
**Pull request by timlinux**
## Description

Three new easter eggs:

1) world : Add a world map to your canvas - it is the same world map as used by the CRS chooser
2) contributors : Adds the contributors json (used to be in docs, now moved to resources/data)to the map canvas so you can quickly see the locations of QGIS developers in QGIS
3) hackfests : Adds qgis_hackfests.json (in resources/data) to the canvas so you can quickly see the locations of all historical QGIS hackfests. Note the attribute table has other interesting data too.

I tried to make some nice cartography for the contributors and hackfests:

gis

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [ ] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [ ] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: src/app/qgsstatusbarcoordinateswidget.cpp [hunk 1]
```diff
@@ -117,7 +119,19 @@ void QgsStatusBarCoordinatesWidget::validateCoordinates()
   {
     return;
   }
-  if ( mLineEdit->text() == QLatin1String( "dizzy" ) )
+  else if ( mLineEdit->text() == QLatin1String( "world" ) )
+  {
+    this->world();
```
Comment (nyalldawson) [inline src/app/qgsstatusbarcoordinateswidget.cpp]: Remove this-> prefix (elsewhere too)

File: src/app/qgsstatusbarcoordinateswidget.cpp [hunk 2]
```diff
@@ -194,6 +208,52 @@ void QgsStatusBarCoordinatesWidget::dizzy()
   mMapCanvas->setTransform( matrix );
 }
 
+void QgsStatusBarCoordinatesWidget::contributors()
+{
+  if ( !mMapCanvas )
+  {
+    return;
+  }
+  QString myFileName = QgsApplication::pkgDataPath() + QStringLiteral( "/resources/data/contributors.json" );
```
Comment (nyalldawson) [inline src/app/qgsstatusbarcoordinateswidget.cpp]: Remove ""my" prefixes

File: src/app/qgsstatusbarcoordinateswidget.cpp [hunk 3]
```diff
@@ -194,6 +208,52 @@ void QgsStatusBarCoordinatesWidget::dizzy()
...
+  QFileInfo myFileInfo = QFileInfo( myFileName );
+  QgsVectorLayer *layer = new QgsVectorLayer( myFileInfo.absoluteFilePath(),
+      "QGIS Contributors", QStringLiteral( "ogr" ) );
```
Comment (nyalldawson) [inline src/app/qgsstatusbarcoordinateswidget.cpp]: Should this be translated?

File: src/app/qgsstatusbarcoordinateswidget.cpp [hunk 4]
```diff
@@ -194,6 +208,52 @@ void QgsStatusBarCoordinatesWidget::dizzy()
...
+  // Register this layer with the layers registry
+  QgsProject::instance()->addMapLayer( layer );
+  layer->setAutoRefreshInterval( 500 );
+  layer->setAutoRefreshEnabled( true );
```
Comment (nyalldawson) [inline src/app/qgsstatusbarcoordinateswidget.cpp]: What's this for? The layer data shouldn't be changing, unless I'm missing something?

Comment (Gustry) [reply src/app/qgsstatusbarcoordinateswidget.cpp]: The data no, but the style yes, cf https://twitter.com/timlinux/status/1033784351740198913

**Comment by geomenke** [comment]
Will these be activated by typing those keywords into the coordinate box as in the 'bored' easter egg?

File: src/app/qgsstatusbarcoordinateswidget.cpp (hunk 1 above)
Comment (timlinux) [reply src/app/qgsstatusbarcoordinateswidget.cpp]: Hah my python alter ego feels there should be 'this' or 'self' behind every call :-P ... will do...

File: src/app/qgsstatusbarcoordinateswidget.cpp (hunk 4 above)
Comment (timlinux) [reply src/app/qgsstatusbarcoordinateswidget.cpp]: @nyalldawson yes the little wheel icons rotate pleasingly on the screen because of that....

**Comment by timlinux** [comment]
@geomenke wrote:

> Will these be activated by typing those keywords into the coordinate box as in the 'bored' easter egg?

Yes correct!

**Comment by nyalldawson** [comment]
Looks good now!

**Comment by timlinux** [comment]
@nyalldawson 

> Looks good now!

Thanks for reviewing!

**Comment by nirvn** [comment]
@timlinux , this PR broke `make install` on linux (maybe other packages too):

```
CMake Error at doc/cmake_install.cmake:41 (file):
  file INSTALL cannot find
  "/home/webmaster/dev/cpp/QGIS/doc/contributors.json".
Call Stack (most recent call first):
  cmake_install.cmake:58 (include)
```

**Comment by luipir** [comment]
the docker image used by travis does not  install qgis with make install?

**Comment by timlinux** [comment]
@nirvn ah I can fix that - will do so ina minute

**Comment by rduivenvoorde** [comment]
@timlinux working for me here on recent master, self compiled Debug. 
Minor things: debug info keeps running (a lot of renderer debug info). Also when you do 'bored' after that, and the actual animation is frozen.
I started with 'hackfests' and was missing the world map. Would it be too much to always add the map when either hackfests or contributors is typed?

**Pull request merged**